urlshort/handler.go

```diff
@@ -11,8 +14,19 @@ import (
 // If the path is not provided in the map, then the fallback
 // http.Handler will be called instead.
 func MapHandler(pathsToUrls map[string]string, fallback http.Handler) http.HandlerFunc {
-	//	TODO: Implement this...
-	return nil
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		path := r.URL.Path
+		redirectUrl, ok := pathsToUrls[path]
+		if ok {
+			log.Println("Before custom with " + redirectUrl)
+			http.Redirect(w, r, redirectUrl, http.StatusSeeOther)
+			log.Println("After custom")
+		} else {
+			log.Println("Before")
+			fallback.ServeHTTP(w, r) // call original
+			log.Println("After")
+		}
```

Comment:
Another more compact way this could be written (not necessarily better) is:
```go
	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		if redirectUrl, ok := pathsToUrls[r.URL.Path]; ok {
			http.Redirect(w, r, redirectUrl, http.StatusSeeOther) // I'm not sure this is the right status
			return
		} 
		fallback.ServeHTTP(w, r) // call fallback
	})
```

Reply:
OK